### PR TITLE
add API to set mastering display, target display peak brightness

### PR DESCRIPTION
**Pull request by ram-mohan**
add API to set mastering display, target display peak brightness

### CODE REVIEW

**Comment by ram-mohan** [comment]
This is not a complete change as java interfaces also requires to be updated if we are introducing this API. This is added to get some comments from reviewers.

**Comment by ram-mohan** [comment]
> This is not a complete change as java interfaces also requires to be updated if we are introducing this API. This is added to get some comments from reviewers.

java interfaces for this api are also added.

**Review by harishdm — APPROVED** [review]

File: lib/src/jpegr.cpp [hunk 1]
```diff
@@ -498,6 +499,22 @@ uhdr_error_info_t JpegR::convertYuv(uhdr_raw_image_t* image, uhdr_color_gamut_t
   return status;
 }
 
+float JpegR::getMasteringDisplayMaxLuminance(uhdr_color_transfer_t transfer) {
+  switch (transfer) {
+    case UHDR_CT_LINEAR:
+      return mMaxDispBrightness;
+    case UHDR_CT_HLG:
+      return mMaxDispBrightness != -1.0f ? mMaxDispBrightness : kHlgMaxNits;
+    case UHDR_CT_PQ:
```
Comment (DichenZhang1) [inline lib/src/jpegr.cpp:508]: Does it make sense to make QP max brightness also configurable?

File: lib/src/jpegr.cpp [hunk 2]
```diff
@@ -1479,16 +1501,17 @@ uhdr_error_info_t JpegR::applyGainMap(uhdr_raw_image_t* sdr_intent, uhdr_raw_ima
 #endif
           }
 
-          rgb_hdr = rgb_hdr / display_boost;
           size_t pixel_idx = x + y * dest->stride[UHDR_PLANE_PACKED];
 
           switch (output_ct) {
             case UHDR_CT_LINEAR: {
+              rgb_hdr = rgb_hdr / gainmap_metadata->hdr_capacity_max;
```
Comment (DichenZhang1) [inline lib/src/jpegr.cpp:1508]: I feel with the logic the input `max_display_boost` will become unused in this pass. Also the logic in the OpenGL pass isn't updated

Comment (ram-mohan) [reply lib/src/jpegr.cpp]: Thank you. I have addressed this.

File: lib/src/jpegr.cpp (hunk 1 above)
Comment (ram-mohan) [reply lib/src/jpegr.cpp:510]: Thank you. I have addressed this.

**Comment by gregbenz** [comment]
Thank you for adding support, this is very helpful. Results look great.

**Review by DichenZhang1 — APPROVED** [review]

**Pull request merged**